rdflib/term.py

```diff
@@ -1397,6 +1400,14 @@ def _writeXML(xmlnode):
     _XSD_DECIMAL,
 )
 
+# these are not guranteed to sort because it is not possible
+# to calculate a total order over all valid members of the type
+# the function must partition the type into subtypes that do have total orders
+_NO_TOTAL_ORDER_TYPES = {
+    datetime:lambda value:bool(value.tzinfo),
+    time:lambda value:bool(value.tzinfo),
```

Comment:
this is actually quite bad, as it will never sort (date)time objects anymore... fix is incoming...

Reply:
Are you sure? I have a test for this which produces the expected and correct reordering.

https://github.com/tgbugs/pyontutils/blob/master/test/nasty.ttl#L203
https://github.com/tgbugs/pyontutils/blob/master/test/good.ttl#L234

Reply:
Remember that this is a partition function not the sort key.

Reply:
Hrm... checking again, it does seem that I actually did not implement this correctly, which makes it all the more curious why my tests pass.

Reply:
already fixed, just see #862

Reply:
but thanks for the reply ;)

Reply:
That is the right solution and I think ultimately much clearer than what I was thinking about, which was to turn the caster into just the partition, which I think is where my brain fart came from. Many thanks!